airbyte-local.sh

```diff
@@ -406,6 +453,238 @@ function loadState() {
+          name: airbyte-config
+      command:
+        - sh
+        - -c
+        - |
+          mkfifo /pipes/src_out
+          mkfifo /pipes/dst_in
+          mkfifo /pipes/state
+          mkfifo /pipes/discover
+          ITERATION=0
+          MAX_ITERATION=600
+          DISK_USAGE=\$(du -s /config | awk '{print \$1;}')
+
+          until [ -f FINISHED_UPLOADING -o \$ITERATION -ge \$MAX_ITERATION ]; do
+            ITERATION=\$((ITERATION+1))
+            LAST_DISK_USAGE=\$DISK_USAGE
+            DISK_USAGE=\$(du -s /config | awk '{print \$1;}')
+            if [ \$DISK_USAGE -gt \$LAST_DISK_USAGE ]; then
+              ITERATION=0
+            fi
+            sleep 0.5
+          done
+
+          if [ -f FINISHED_UPLOADING ]; then
+            echo "All files copied successfully, exiting with code 0..."
+            exit 0
+          else
+            echo "Timeout while attempting to copy to init container, exiting with code 1..."
+            exit 1
+          fi
+  containers:
+    - name: source
+      image: $SRC_DOCKER_IMAGE_PLACEHOLDER
+      workingDir: "/config"
+      env:
+        - name: SOURCE_CONFIG
+          value: source_config.json
+        - name: SOURCE_CATALOG
+          value: source_catalog.json
+        - name: INPUT_STATE
+          value: input_state.json
+      command:
+        - sh 
+        - -c
+        - |
+          if [[ ! -s "\$SOURCE_CATALOG" ]]; then
+            echo "Generating source catalog"
+            ((eval "\$AIRBYTE_ENTRYPOINT discover --config \${SOURCE_CONFIG}") | tee /pipes/discover | grep -v '"CATALOG"') &
+            CHILD_PID=\$!
+            wait \$CHILD_PID
+            echo "Completed generating source catalog"
+            ITERATION=0
+            MAX_ITERATION=10
+            until [ -s "\$SOURCE_CATALOG" -o \$ITERATION -ge \$MAX_ITERATION ]; do
+              ITERATION=\$((ITERATION+1))
+              echo "Waiting for non-empty file \$SOURCE_CATALOG"
+              sleep 0.5
+            done
+            if [[ ! -s "\$SOURCE_CATALOG" ]]; then
+              echo "Timed out waiting for source catalog \"\$SOURCE_CATALOG\""
+              exit 1
+            fi
+          fi
+          echo "Starting source connector"
+          ((eval "\$AIRBYTE_ENTRYPOINT read --config \${SOURCE_CONFIG} --catalog \${SOURCE_CATALOG} --state \${INPUT_STATE}" 2>&1 | tee /pipes/src_out) | grep -v '"type":"RECORD"' | grep -v '"type":"STATE"') &
+          CHILD_PID=\$!
+
+          echo "Source connector: Waiting on CHILD_PID \$CHILD_PID"
+          wait \$CHILD_PID
+          EXIT_STATUS=\$?
+          echo "Source connector EXIT_STATUS: \$EXIT_STATUS"
+          exit \$EXIT_STATUS
+      volumeMounts:
+        - name: pipes-volume
+          mountPath: /pipes
+        - mountPath: /config
+          name: airbyte-config
+      # resources:
+      #   limits:
+      #     cpu: "3"
+      #     memory: 256Mi
+      #   requests:
+      #     cpu: 500m
+      #     memory: 50Mi
```

Comment:
why these are commented out? what are the defaults then? (same for other resources section below)

Reply:
I wanted to give it as an example - there is a lot of parameter to set there. Anyhow - I added 2 params (see latest commit) - for mem and cpu limit and they can now be set on command line - for source and destination containers only. The init and transformer container just pipe data and should not require special limits. Their limits/requests will assume whatever is default on the cluster.

Reply:
Who is this example for? one of us? we can figure it out once we need it. let's only set the necessary args

Reply:
all that is removed already. Now allow setting cpu and mem limits with params on command line